src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinder.java

```diff
@@ -59,8 +60,17 @@
     static /* not final */ boolean USE_HEAVYWEIGHT_CHECKOUT = Boolean.getBoolean(SCMBinder.class.getName() + ".USE_HEAVYWEIGHT_CHECKOUT"); // TODO 2.4+ use SystemProperties
     private String scriptPath;
 
+    public Object readResolve() {
+        this.scriptPath = WorkflowBranchProjectFactory.SCRIPT;
+        return this;
+    }
+
     public SCMBinder(String scriptPath) {
-        this.scriptPath = scriptPath;
+        if (StringUtils.isEmpty(scriptPath)) {
```

Comment:
This is not called from user input so it can (and should) assume that `scriptPath` is exact as passed.